src/nvim/api/vim.c

```diff
@@ -754,11 +754,15 @@ Object nvim_get_option_value(String name, Dict(option) *opts, Error *err)
 /// |:set|: for global-local options, both the global and local value are set
 /// unless otherwise specified with {scope}.
 ///
+/// Note the options {scope}, {win} and {buf} cannot be used together.
+///
 /// @param name      Option name
 /// @param value     New option value
 /// @param opts      Optional parameters
 ///                  - scope: One of 'global' or 'local'. Analogous to
 ///                  |:setglobal| and |:setlocal|, respectively.
+///                  - win: |window-ID|. Set window local option.
+///                  - buf: Buffer number. Set buffer local option.
```

Comment:
- use double quotes in docs (single quotes are links to options)
- could save the "exclusive" stuff by this interface:

```suggestion
///                  - scope: One of "global", "local", "buf" or "win". 
///                  |:setglobal| |:setlocal|. If "buf" or "win", then 
///                  - id: buffer number or |window-ID| when scope=buf or scope=win.
```

Reply:
shouldn't it be `{id}` instead of `"id"`

Reply:
This won't work as window options can be local, global or neither. The rules are weird and I myself have only just finally understood them. See discussion in #18524

Reply:
In theory we could even allow to change the saved value for a particular buffer-window combination, so "buf" and "win" keys must be separate (there is no pressing need to support this right now, but it would be a mistake to paint us into a corner were we cannot).

Reply:
good points -- retract my suggestion!